catalyst/contrib/datasets/mnist.py

```diff
@@ -209,19 +209,74 @@ def extra_repr(self):
 
 class MnistMLDataset(MetricLearningTrainDataset, MNIST):
     """
-    Simple wrapper for MNIST dataset
+    Simple wrapper for MNIST dataset for metric learning train stage.
+    This dataset can be used only for training. For test stage
+    use MnistQGDataset.
+
+    For this dataset we use only training part of the MNIST and only
+    those images that are labeled as 0, 1, 2, 3, 4.
     """
 
+    _split = 5
+    classes = [
+        "0 - zero",
+        "1 - one",
+        "2 - two",
+        "3 - three",
+        "4 - four",
+    ]
+
+    def __init__(self, root: str, train: bool = True, **kwargs):
+        """
+
+        Args:
+            root: root directory of the dataset
+            train: for MnistMLDataset should always be True
+
+        Raises:
```

Comment:
I think we should remove `train` from args. and consider 2 situations:
`train` is in `kwargs`, then we should check that train is True, otherwise raise an Error 
`train` is not in `kwargs`, then we should add it as `train=True`

in current implementation we allow to pass `train` value directly to init, but don't allow to change what, this is confusing behaviour

Reply:
what is you concern about? 
something like this?
`kwargs = {"train": False}
dataset = MnistMLDataset("./data", True, **kwargs)`
it will fail with exception about multiple values for train args.

I thougt it to be a good idea to remind in docs that train should always be True in this case 🤔

Reply:
oh, i think i understood, ok